scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+def get_repository_commit_sha(repository_location):
+    command = f"git -C {repository_location} rev-parse HEAD"
```

Comment:
We probably need also take care of spaces in the repository_location. Specifying a command as: `['git', '-C', repository_location, 'rev-parse', 'HEAD']` might do the job.